### PR TITLE
Test remat graph

### PR DESCRIPTION
**Pull request by Chenqll**
在 VM = 1 时，DTR 在静态图下运行的测试文件

### CODE REVIEW

**Comment by Chenqll** [comment]
# Backward in Resnet18 报错
## 1. 脚本，test5 和 test6 分别是 resnet 的 eager 和 graph 模式跑完整训练的代码，单独运行 test6 不会有任何输出，在 CPU 下的 test 6 也会有 print 的的输出。
```python
import oneflow as flow
import oneflow.nn as nn
import flowvision

flow.remat.set_budget("300MB")
flow.remat.set_small_pieces_optimization(False)
device='cpu+remat'


loss5=[]
def test5():
    model=flowvision.models.resnet18().to(device)
    criterion = nn.CrossEntropyLoss().to(device)
    optimizer=flow.optim.SGD(model.parameters(),lr=0.1,momentum=0)
    for p in model.parameters():
        p.grad = flow.zeros_like(p).to(device)
    x = flow.rand(10, 3, 224, 224).to(device)
    target = (
        flow.randint(low=0, high=1000, size=(x.shape[0],)).to(device).to(flow.int32)
    )
    flow._oneflow_internal.remat.disable_eviction(x)
    flow._oneflow_internal.remat.disable_eviction(target)
    for _ in range(10):
        output = model(x)
        loss = criterion(output, target)
        loss.backward()
        loss5.append(loss)
        del loss
        optimizer.step()
        optimizer.zero_grad()



loss6=[]
def test6():
    model=flowvision.models.resnet18().to(device)
    criterion = nn.CrossEntropyLoss().to(device)
    optimizer=flow.optim.SGD(model.parameters(),lr=0.1,momentum=0)
    for p in model.parameters():
        p.grad = flow.zeros_like(p).to(device)
    x = flow.rand(10, 3, 224, 224).to(device)
    target = (
        flow.randint(low=0, high=1000, size=(x.shape[0],)).to(device).to(flow.int32)
    )

    flow._oneflow_internal.remat.disable_eviction(x)
    flow._oneflow_internal.remat.disable_eviction(target)

    class GraphMyBackward(nn.Graph):
        def __init__(self):
            super().__init__()
            self.model=model
            self.loss_fn=criterion
            self.add_optimizer(optimizer)
        def build(self,x,y):
            y_pred=self.model(x)
            loss=self.loss_fn(y_pred,y)
            loss.backward()
            optimizer.step()
            return loss
        
    graph=GraphMyBackward()
    for _ in range(10):
        loss = graph(x,target)
        loss6.append(loss)

test5()
print("loss5: \n ",loss5)
print("recomputation1 num0 ", flow._oneflow_internal.remat.recomputation_num())

test6()
print("loss6: \n ",loss6)
print("recomputation1 num: ", flow._oneflow_internal.remat.recomputation_num())
```
## 2. 只有在同时运行 test5 test 6 时， 才会报下面的错误：
loss5: 
  [tensor(6.9992, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(3.3670, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(1.7277, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(1.1734, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(0.9022, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(0.3231, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(0.1123, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(0.0395, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(0.0271, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>), tensor(0.0222, dtype=oneflow.float32, grad_fn=<broadcast_div_backward>)]
recomputation1 num0  20
F20230414 11:36:49.082909 3619229 eager_blob_object.cpp:119] Check failed: 'ptr' Must be non NULL 
*** Check failure stack trace: ***
    @     0x7f97f83bd5dc  google::LogMessage::Fail()
    @     0x7f97f83bd517  google::LogMessage::SendToLog()
    @     0x7f97f83bce02  google::LogMessage::Flush()
    @     0x7f97f83c095e  google::LogMessageFatal::~LogMessageFatal()
    @     0x7f97d6dd4185  google::CheckNotNull<>()
    @     0x7f97d6dd3001  oneflow::vm::EagerBlobObject::raw_dptr()
    @     0x7f97f1caf84a  oneflow::vm::EagerBlobObject::mut_raw_dptr()
    @     0x7f97f1e1e721  oneflow::user_op::Tensor::mut_dptr<>()
    @     0x7f97da2e2594  oneflow::NormalizationTrainCpuKernel<>::Compute()
    @     0x7f97f20f173c  oneflow::user_op::OpKernel::Compute()
    @     0x7f97da756d3d  oneflow::one::StatefulOpKernel::Compute()
    @     0x7f97d98dc28f  oneflow::vm::OpCallInstructionUtil::OpKernelCompute()
    @     0x7f97d98daaee  oneflow::vm::OpCallInstructionUtil::Compute()
    @     0x7f97d98dc2dd  oneflow::vm::OpCallInstructionUtil::ComputeFnForRemat()
    @     0x7f97d98dcf1b  std::_Function_handler<>::_M_invoke()
    @     0x7f97d991005b  std::function<>::operator()()
    @     0x7f97d990d4c5  oneflow::vm::RematHelper::RematInputs()
    @     0x7f97d98da710  oneflow::vm::OpCallInstructionUtil::Compute()
    @     0x7f97d98dc2dd  oneflow::vm::OpCallInstructionUtil::ComputeFnForRemat()
    @     0x7f97d98dcf1b  std::_Function_handler<>::_M_invoke()
    @     0x7f97d991005b  std::function<>::operator()()
    @     0x7f97d990d4c5  oneflow::vm::RematHelper::RematInputs()
    @     0x7f97d98da710  oneflow::vm::OpCallInstructionUtil::Compute()
    @     0x7f97d98dc2dd  oneflow::vm::OpCallInstructionUtil::ComputeFnForRemat()
    @     0x7f97d98dcf1b  std::_Function_handler<>::_M_invoke()
    @     0x7f97d991005b  std::function<>::operator()()
    @     0x7f97d990d4c5  oneflow::vm::RematHelper::RematInputs()
    @     0x7f97d98da710  oneflow::vm::OpCallInstructionUtil::Compute()
    @     0x7f97d98dc2dd  oneflow::vm::OpCallInstructionUtil::ComputeFnForRemat()
    @     0x7f97d98dcf1b  std::_Function_handler<>::_M_invoke()
    @     0x7f97d991005b  std::function<>::operator()()
    @     0x7f97d990d4c5  oneflow::vm::RematHelper::RematInputs()
